docs/Radius_Authentication.md

```diff
@@ -0,0 +1,83 @@
+
+## Prerequisites
+
+The following conditions are assumed by this process:
+
+* Your nodes are running on Ubuntu 14.04.
+* Your nodes have access to both Clearwater and Ubuntu repositories.
+* Your SSH configuration allows password authentication and PAM (the correct configuration will be detailed below).
+* You have access to a third-party RADIUS server (such as freeRADIUS).
+* Your firewall must allow UDP traffic to the above server on port 1812.
+
+## Installation
+
+### Package installation
+
+Install the Clearwater RADIUS authentication package:
+
+    sudo apt-get install clearwater-radius-auth
+
+### Configuration
+
+The details of your RADIUS server will need to be entered into `/etc/pam_radius_auth.conf`. This file provides an example of how entries should be structured:
+* Multiple entries are allowed, but each must be on a new line.
+* Each line consists of three fields:
+    * `server[:port]` (The default is port 1812. All traffic will be UDP)
+    * `secret`
+    * `[timeout]` (Default timeout is 3 seconds)
+* The secret is shared between each client and the server to allow simple encryption of passwords. The secret must match the entry for the client in the RADIUS server configuration.
+* Both the port and timeout entries are optional.
+
+Your sshd configuration must allow password authentication, and use of PAM. If you are unsure, check that the `PasswordAuthentication` and `UsePAM` entries in `/etc/ssh/sshd_config` are set to `yes`. Any changes to ssh configuration will require the ssh process to be restarted before coming into effect.
+
+You must ensure that your firewall/security groups allow UDP traffic to the RADIUS server on port 1812. 
+
+## Usage
+
+Once the above is installed and configured, any user provisioned in the RADIUS server can attempt SSH or SFTP access to the configured node, and on providing their password they should be authenticated and logged in, acting as the default Ubuntu user. Commands such as `who` or `last` will output the username supplied at login, and this will also be recorded in the auth log `/var/log/auth.log`.
+
+Any users provisioned locally on the node should see no change to their authentication experience. By default, RADIUS authentication is set to be a sufficient, but not required condition. As such failing to authenticate against the server credentials will cause the authentication attempt to fall back to checking locally provisioned details. This can be altered, but is not something we support. See below for further details on configuration options.
+
+## Common issues
+
+* If you are not seeing any traffic reaching your RADIUS server, and entries in `/var/log/auth.log` state that no RADIUS server was reachable, re-check the rADIUS server entry in `/etc/pam_radius_auth.conf`, and ensure that your firewall is configured to allow UDP traffic to the RADIUS server on port 1812.
```

Comment:
Is this useful troubleshooting? It feels a bit like 'If you're seeing errors, make sure that you've configured the system correctly"